__tests__/__unit__/dataset/actions.unit.test.ts

```diff
@@ -0,0 +1,1508 @@
+    Object.defineProperty(vscode.window, "showQuickPick", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.commands, "executeCommand", { value: jest.fn(), configurable: true });
+    Object.defineProperty(globals, "LOG", { value: jest.fn(), configurable: true });
+    Object.defineProperty(globals.LOG, "debug", { value: jest.fn(), configurable: true });
+    Object.defineProperty(globals.LOG, "error", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe, "Download", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe.Download, "dataSet", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe, "Delete", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe.Delete, "dataSet", { value: jest.fn(), configurable: true });
+    Object.defineProperty(fs, "unlinkSync", { value: jest.fn(), configurable: true });
+    Object.defineProperty(fs, "existsSync", { value: jest.fn(), configurable: true });
+    Object.defineProperty(sharedUtils, "concatChildNodes", { value: jest.fn(), configurable: true });
+    Object.defineProperty(Profiles, "getInstance", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe, "List", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe.List, "dataSet", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode, "ProgressLocation", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "createWebviewPanel", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.env, "clipboard", { value: clipboard, configurable: true });
+}
+
+// Idea is borrowed from: https://github.com/kulshekhar/ts-jest/blob/master/src/util/testing.ts
+const mocked = <T extends (...args: any[]) => any>(fn: T): jest.Mock<ReturnType<T>> => fn as any;
+
+describe("Dataset Actions Unit Tests - Function createMember", () => {
+    let blockMocks;
+
+    function createBlockMocks() {
+        const session = generateISession();
+        const imperativeProfile = generateIProfile();
+        const zosmfSession = generateBasicZosmfSession(imperativeProfile);
+        const treeView = generateTreeView();
+        const datasetSessionNode = generateDatasetSessionNode(session, imperativeProfile);
+        const mvsApi = generateMvsApi(imperativeProfile);
+        bindMvsApi(mvsApi);
+
+        return {
+            session,
+            zosmfSession,
+            treeView,
+            imperativeProfile,
+            datasetSessionNode,
+            mvsApi,
+            testDatasetTree: generateDatasetTree(datasetSessionNode, treeView)
+        };
+    }
+
+    afterAll(() => jest.restoreAllMocks());
+
+    it("Checking of common dataset member creation", async () => {
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const parent = new ZoweDatasetNode("parent", vscode.TreeItemCollapsibleState.Collapsed,
```

Comment:
The same node is used in all it() blocks, can it be declared inside createBlockMocks()?

Reply:
I think I agree in this particular case, because the use of is really identical. But generally I do not mind when stuff is repeated among individual tests. Every test is a unit of its own and trying to extract too much into helper functions creates indirections can make it harder to read the tests in the end.

Reply:
What might help though is spread the `blockckMock` like `const { session, msApi, parent} = createBlockMocks();` But that is a topic for further PRs, not here.

Reply:
This was just an idea and Vit explained to me that if nodes change slightly in later tests, the node would have to be redeclared anyway. So this is fine, feel free to mark resolved if you want ^^